File: trunk/JSI2/src/main/org/xidea/el/parser/ExpressionToken.java
package org.xidea.el.parser;

/**
 * @author jindw
 */
public abstract interface ExpressionToken {

	public static final int VALUE_CONSTANTS = 0;//'c';
	public static final int VALUE_VAR = 1;//'n';
	public static final int VALUE_NEW_MAP = 2;//{;
	public static final int VALUE_NEW_LIST = 3;//[;
	public static final int VALUE_LAZY = 4;
	
	public static final int BRACKET_BEGIN = 5;//'('[{;
	public static final int BRACKET_END = 6;//')']};


	
	//与正负符号共享字面值
	public static final int OP_ADD = 10;//'+';
	public static final int OP_SUB = 11;//'-';
	
	public static final int OP_MUL = 12;//'*';
	public static final int OP_DIV = 13;//'/';
	public static final int OP_MOD = 14;//'%';
	public static final int OP_QUESTION = 15;//'?';
	public static final int OP_QUESTION_SELECT = 16;//':';
	public static final int OP_GET_PROP = 17;//'.';
	
	public static final int OP_LT = 18;//'<';
	public static final int OP_GT = 19;//'>';
	public static final int OP_LTEQ = 20;//('<' << 8) + '=';// '<=';
	public static final int OP_GTEQ = 21;//('>' << 8) + '=';// '>=';
	public static final int OP_EQ = 22;//('=' << 8) + '=';// '==';
	public static final int OP_NOTEQ = 23;//('!' << 8) + '=';// '!=';
	public static final int OP_AND = 24;//('&' << 8) + '&';
	public static final int OP_OR = 25;//('|' << 8) + '|';// '||';
	
	

	public static final int OP_NOT = 26;//'!';
	public static final int OP_POS = 27;//('+'<<8) + 'p';//負數
	public static final int OP_NEG = 28;//('-'<<8) + 'n';//負數

	public static final int OP_GET_METHOD = 29;//('.'<<16) + ('('<<8) + ')';
	public static final int OP_GET_GLOBAL_METHOD = 30;//('('<<8) + ')';
	public static final int OP_INVOKE_METHOD = 31;


	//与Map Join 共享字面量（map join 会忽略）
	public static final int OP_PARAM_JOIN = 32;//,
	//与三元运算符共享字面值
	public static final int OP_MAP_PUSH = 33;//:,
	
	
	public abstract int getType();

	public abstract String toString();
	


}

File: src/python/org/xidea/el/parser/ExpressionToken.py
#!/usr/bin/env python
# -*- coding: utf-8 -*-
VALUE_CONSTANTS = 0
VALUE_VAR = 1
VALUE_NEW_MAP = 2
VALUE_NEW_LIST = 3
VALUE_LAZY = 4
BRACKET_BEGIN = 5
BRACKET_END = 6
OP_ADD = 10
OP_SUB = 11
OP_MUL = 12
OP_DIV = 13
OP_MOD = 14
OP_QUESTION = 15
OP_QUESTION_SELECT = 16
OP_GET_PROP = 17
OP_LT = 18
OP_GT = 19
OP_LTEQ = 20
OP_GTEQ = 21
OP_EQ = 22
OP_NOTEQ = 23
OP_AND = 24
OP_OR = 25
OP_NOT = 26
OP_POS = 27
OP_NEG = 28
OP_GET_METHOD = 29
OP_GET_STATIC_METHOD = 30
OP_INVOKE_METHOD = 31
OP_PARAM_JOIN = 32
OP_MAP_PUSH = 33
class ExpressionToken(object):
    """ generated source for ExpressionToken
    """
    def getType(self):
        raise NotImplementedError()

    def toString(self):
        raise NotImplementedError()



class VarToken(ExpressionToken):
    """ generated source for VarToken

    """
    value = ""

    def __init__(self, value):
        self.value = self.value

    def getType(self):
        return VALUE_VAR

    def getValue(self):
        return self.value

    def toString(self):
        return self.value


